ui/articles-aem-renderer/src/androidTest/java/org/cru/godtools/articles/aem/db/ArticleDBTest.java

```diff
@@ -0,0 +1,145 @@
+package org.cru.godtools.articles.aem.db;
+
+import android.arch.lifecycle.Lifecycle;
+import android.arch.lifecycle.LifecycleOwner;
+import android.arch.lifecycle.Observer;
+import android.arch.persistence.room.Room;
+import android.content.Context;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.cru.godtools.articles.aem.model.Article;
+import org.cru.godtools.articles.aem.model.Attachment;
+import org.cru.godtools.articles.aem.model.ManifestAssociation;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+
+import timber.log.Timber;
+
+/**
+ * Instrumented test, which will execute on an Android device.
+ *
+ * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
+ */
+@RunWith(AndroidJUnit4.class)
+public class ArticleDBTest implements LifecycleOwner {
+
+    private ArticleDao mArticleDao;
+    private AttachmentDao mAttachmentDao;
+    private ManifestAssociationDao mAssociationDao;
+    private ArticleRoomDatabase db;
+
+    private Observer<List<Article>> articleObserver;
+
+    @Before
+    public void createDb() {
+        Context context = InstrumentationRegistry.getTargetContext();
+        db = Room.inMemoryDatabaseBuilder(context,
+                ArticleRoomDatabase.class).allowMainThreadQueries().build();
+        mArticleDao = db.articleDao();
+        mAttachmentDao = db.attachmentDao();
+        mAssociationDao = db.manifestAssociationDao();
+
+    }
+
+    @After
+    public void closeDb() {
+        db.close();
+    }
+
+    @Test
+    public void useAppContext() {
+        // Context of the app under test.
+        Context appContext = InstrumentationRegistry.getTargetContext();
+
+        List<Article> articles = new ArrayList<>();
+
+        for (int i = 0; i < 12; i++) {
+            Article article = new Article();
+
+            article.mContent = "<p> The Body </>";
+            try {
+                article.mDateCreated = new SimpleDateFormat("E MMM dd yyyy HH:mm:ss zz")
+                        .parse("Fri Jun 08 2018 18:55:00 GMT+0000").getTime();
+                article.mDateUpdated = new SimpleDateFormat("E MMM dd yyyy HH:mm:ss zz")
+                        .parse("Sat May 19 2018 00:23:39 GMT+0000").getTime();
+            } catch (ParseException e) {
```

Comment:
within test classes you shouldn't be catching any exceptions, you should let them throw so that the test actually fails if something goes wrong.
The only time you would catch an Exception is if it is expected, and then you should `fail()` within the test body so you know if the Exception wasn't actually thrown when you were expecting it to throw.

Reply:
for example:
```java
    @Test
    public void verifyMissingService() throws Exception {
        try {
            controller.getTicket(null, ticketParams(ACCESS_TOKEN, null));
            fail();
        } catch (final OAuthApiError e) {
            assertThat(e, matchesApiError(BAD_REQUEST, ERROR_BEARER_INVALID_REQUEST));
        }
    }
```